### PR TITLE
Is Agile always the best approach blog post

### PR DESCRIPTION
**Pull request by dogle-scottlogic**
Please add a direct link to your post here:

[https://<your-username>.github.io/blog/<post-url>](https://dogle-scottlogic.github.io/blog/2024/04/17/is-agile-the-answer.html)

Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [x] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm spellcheck` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [x] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2024-04-17-is-agile-the-answer.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Agile and teamwork
+
+If we take a look at the [Agile Manifesto](https://agilemanifesto.org/) again it is, like I said earlier, just some general advice on what to focus on when developing software. I believe that the Agile Manifesto and Agile development in general has strong ties to teamwork, good Agile teams are good teams in general and Agile is a key part of facilitating that teamwork. In my opinion great teams have the following properties, strong communication coupled with a clear hierarchical structure which utilises delegation to promote and facilitate independent, creative thought and initiative. Let’s break that down a little bit, if we look at the items on the left of the Agile Manifesto, two of the four ("Individuals and interactions" and "Customer collaboration") are around communication, along with some of the principles. A lot of the common ceremonies we associate with Agile are built around the idea of facilitating better communication both within the core team and with others, such as stand-up and sprint demo.
```
Comment (MattCline16) [inline _posts/2024-04-17-is-agile-the-answer.md]: `[...] have the following properties:`

File: _posts/2024-04-17-is-agile-the-answer.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Why Agile can fail
+
+Why does Agile fail then? Probably for a multitude of reasons, but I suspect that, ultimately, one of the main reasons is a failure to adapt processes, i.e a failure of the team to modify their ways of working appropriately to suit the current circumstances. This may manifest in various ways, such as teams sticking religiously to processes and ceremonies that are not adding value or worse, actively detrimental to progress and morale. It may be that teams are too cautious about trying new things and new approaches or that they are actively blocked from doing so through not having true ownership of the processes. They may be struggling to identify the things that are causing issues due to a lack of regular reflection or because people do not feel empowered to point out where the pain points are. Whatever the reasons, in the end the result will be the same, the development process will falter and team morale will suffer. If the ceremonies and processes that are commonly associated with Agile are seen to be part of the problem, it’s easy to see how Agile as a whole could be given the blame when things go wrong.
```
Comment (MattCline16) [inline _posts/2024-04-17-is-agile-the-answer.md]: I feel like another reason Agile can "fail" is either an attempt to scale up one team's process to a larger number of teams or the push down from a department a prescribed consistent way to do agile. What both of these problems stem from is missing out that the process should be for the specific team and adaptable by that team. As soon as you take the agency to improve team working practices out of the hands of the team you are depriving that team agility over their own processes and very quickly we lose team morale, we chafe against wider processes and we see "agile" as the problem where in fact it is "prescribed agile" that is the problem.

Comment (dogle-scottlogic) [reply _posts/2024-04-17-is-agile-the-answer.md]: yeah that's a really good point - I love the term "prescribed agile", I'll update to try and work that in. Thanks Matt 👍

File: _posts/2024-04-17-is-agile-the-answer.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
...project? No, probably not. However, if the question is "can Agile be applied successfully to most software development projects?" I think yes, in my opinion that still holds true, all things considered. The super important caveat here however is that I believe this is true **only** for a given definition of Agile. That definition being the high level Agile Manifesto definition we talked about earlier, i.e being Agile in the sense of aligning the team's ways of working with the Agile Manifesto and principles. I think that the Agile Manifesto and the twelve Principles behind it remain good solid advice for software development and that the majority of projects can benefit from working in an collaborative, iterative manner, utilising self organising teams who regularly reflect on their ways of working and adjust to suit the current circumstances. I do not think that Kanban, Scrum or any other given frameworks will always be the best approach for most projects, nor do I think that any single rigid process will work well for a team indefinitely. Teams change shape, size and goals regularly and a successful Agile team needs to have both the will and the power to adapt with those changes.
```
Comment (mdsmith-scottlogic) [inline _posts/2024-04-17-is-agile-the-answer.md]: swap:
Is Agile the right fit for any project?

Is Agile the right fit for every project?

**Review by MattCline16 — APPROVED** [review]

File: _posts/2024-04-17-is-agile-the-answer.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Where to start?
+
+This I think leads into where teams can start to struggle implementing Agile. If you are looking to start using Agile Development, it can be very hard to know where to begin. This can be especially true when you or your team has little experience with Agile to begin with. The Agile Manifesto isn't much help here of course, four short statements prioritising one thing over another isn't much of a guide to getting a project off the ground (nor is it meant to be). This is where Scrum can come in, Scum has a handy [guide](https://scrumguides.org/scrum-guide.html), there are certified Scrum Masters to help out and it's widely used in industry. It's easy then to see why it's a popular starting point if you're looking to create a new Agile team without much prior experience.
```
Comment (jmoore-scottlogic) [inline _posts/2024-04-17-is-agile-the-answer.md]: Typo - "scum" -> "scrum"

**Review by jmoore-scottlogic — APPROVED** [review]

**Pull request merged**